shared/src/test/scala/scala/xml/UtilityTest.scala

```diff
@@ -219,4 +219,15 @@ class UtilityTest {
     val x: Elem = <div>{Text("   My name ")}{Text("  is  ")}{Text("  Harry   ")}</div>
     assertEquals(<div>My name is Harry</div>, Utility.trim(x))
   }
+
+  @Test
+  def issue306InvalidUnclosedEntity(): Unit = {
```

Comment:
Maybe the name of the test method should be more specific: this is about character references, not entities in general...